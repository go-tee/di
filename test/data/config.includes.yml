parameters:
  scalar: 1
  list: [1, 2]

services:
  userManager:
    type: '*github.com/go-tee/di/test.UserManager'

includes:
  - config.includes.child1.yml
  - config.includes.child2.yml
