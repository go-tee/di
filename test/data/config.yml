parameters:
  abc: test
  database: none

services:

  Req:
    scope: prototype
    type: '*github.com/gooff/di/test/example.Request'
    returns: 'example.CreateRequest(req)'
    arguments:
      req: "*http.Request"
    imports:
      - 'net/http'

  sendEmail:
    scope: prototype
    type: '*github.com/gooff/di/test/example.SendEmail'
    interface: 'github.com/gooff/di/test/example.EmailSender'
#    returns: 'example.NewSendEmail(@{req})'
    properties:
      From: '"hi@welcome.com"'
      Request: '@{req}'
    arguments:
      req: "*http.Request"
#
#  CustomerWelcome:
#    type: '*github.com/gooff/di/test/example.CustomerWelcome'
#    returns: example.NewCustomerWelcome(@{sendEmail})
#
##  CustomerWelcomeAuto: example.NewCustomerWelcome

extensions:
  navigation: github.com/gooff/di/ext.NavigationExtension

navigation:
  abc:
    test: ok

includes:
  - database.yml
