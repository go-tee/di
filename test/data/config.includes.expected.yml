parameters:
  list: [1, 2, 5, 6]
  me:
    - config.includes.child1.yml
    - config.includes.child2.yml
    - config.includes.grandchild.yml
  scalar: 4
services:
  userManager:
    type: '*github.com/go-tee/di/test.UserManager'
    properties:
      ServiceProperty: simple
